piqe_ocp_lib/tests/resources/test_ocp_projects.py

```diff
@@ -21,157 +23,556 @@ def setup_params(get_kubeconfig):
 
 
 class TestOcpProjects:
+    def __setup(self, setup_params, project_name):
+        """
+        Performs common setup task of creating a project.
+
+        :param setup_params:
+        :param project_name:
+        :return:
+        """
+        project_api_obj = setup_params["project_api_obj"]
+        api_response = project_api_obj.create_a_project(project_name)
+        assert api_response.metadata.name == project_name
+        logger.info("Project : {}, succesfully created".format(api_response.metadata.name))
+
+    def __cleanup(self, setup_params, project_name):
+        """
+        Performs common clean up task of deleting the specified project.
+
+        :param setup_params: Dictionary containing the OcpBase
+        :param project_name:
+        :return:
+        """
+        project_api_obj = setup_params["project_api_obj"]
+        if project_api_obj.does_project_exist(project_name):
+            project_api_obj.delete_a_project(project_name)
+            logger.info("Project : {}, successfully deleted".format(project_name))
+        is_deleted_project_present = project_api_obj.does_project_exist(project_name)
+        assert is_deleted_project_present is False
+
+    def __log_exception_formatted(self, message, exception):
+        """
+        Provides consistent formatted logging of unexpected exceptions.
+
+        :param message: String containing user specified error message.
+        :param exception: The raised Exception
+        :return:
+        """
+        logger.error(message)
+        logger.error("Status: %s" % exception.status)
```

Comment:
nitpick: We should be using the Python3 way for string formatting, as in f"Status: {exception.status}"